keras_cv/models/convnext.py

```diff
@@ -0,0 +1,494 @@
+            return layer([inputs, x])
+        else:
+            layer = layers.Activation("linear", name=name + "_identity")
+            return inputs + layer(x)
+
+    return apply
+
+
+def Head(num_classes=1000, activation="softmax", name=None):
+    """Implementation of classification head of RegNet.
+
+    Args:
+      num_classes: number of classes for Dense layer
+      activation: activation function for Dense layer
+      name: name prefix
+
+    Returns:
+      Classification head function.
+    """
+    if name is None:
+        name = str(backend.get_uid("head"))
+
+    def apply(x):
+        x = layers.GlobalAveragePooling2D(name=name + "_head_gap")(x)
+        x = layers.LayerNormalization(epsilon=1e-6, name=name + "_head_layernorm")(x)
+        x = layers.Dense(num_classes, activation=activation, name=name + "_head_dense")(
+            x
+        )
+        return x
+
+    return apply
+
+
+def ConvNeXt(
+    include_rescaling,
+    depths,
+    projection_dims,
+    drop_path_rate=0.0,
+    layer_scale_init_value=1e-6,
+    include_top=True,
+    weights=None,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
+    classes=1000,
+    classifier_activation="softmax",
+    model_name="convnext",
+):
+    """Instantiates ConvNeXt architecture given specific configuration.
+
+    Args:
+      include_rescaling: whether or not to Rescale the inputs. If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      depths: An iterable containing depths for each individual stages.
+      projection_dims: An iterable containing output number of channels of
+      each individual stages.
+      drop_path_rate: Stochastic depth probability. If 0.0, then stochastic
+        depth won't be used.
+      layer_scale_init_value: Layer scale coefficient. If 0.0, layer scaling
+        won't be used.
+      include_top: Boolean denoting whether to include classification head to
+        the model.
+      weights: one of `None` (random initialization), `"imagenet"` (pre-training
+        on ImageNet-1k), or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      input_tensor: optional Keras tensor (i.e. output of `layers.Input()`).
+      pooling: optional pooling mode for feature extraction when `include_top`
+        is `False`.
+        - `None` means that the output of the model will be the 4D tensor output
+          of the last convolutional layer.
+        - `avg` means that global average pooling will be applied to the output
+          of the last convolutional layer, and thus the output of the model will
+          be a 2D tensor.
+        - `max` means that global max pooling will be applied.
+      classes: optional number of classes to classify images into, only to be
+        specified if `include_top` is True, and if no `weights` argument is
```

Comment:
Since we're not inferring `classes` based on `weights`, I think we should drop this last clause.

I'm making this change for other models in #689